ci/views.py

```diff
@@ -627,6 +652,14 @@ def identity(request, identity):
         add_subscription_form.fields['messageset'] = forms.ChoiceField(
                                                         choices=choices)
 
+        optout_visible = False
+        details = results.get('details', {})
+        addresses = details.get('addresses', {})
+        for address_type, addresses in addresses.items():
+            for address, info in addresses.items():
+                if not info.get("optedout", False):
+                    optout_visible = True
```

Comment:
This bit of code is a bit confusing to me. After look at it for a while, I think it's supposed to set `optout_visible` to `True` if any of the address details do not have an `optedout` key, or that key is `False`.


If that is the case, maybe a cleaner way to do this is with `any` or `all`:

`optout_visible = any((not d.get('optedout') for _, d in addresses.items()))`